contributors/design-proposals/storage/dynamic_volume_limit.md

```diff
@@ -0,0 +1,67 @@
+# Dynamic attached volume limits
+
+## Goals
+
+Currently the number of volumes attachable to a node is either hardcoded or only configurable via an environment variable. Also
+existing limits only apply to well known volume types like EBS, GCE and is not available to all volume plugins.
+
+This proposal enables any volume plugin to specify those limits and also allows same volume type to have different volume
+limits depending on type of node.
+
+## Implementation Design
+
+### Prerequisite
+
+* This feature will be protected by an alpha feature gate, so as API and CLI changes needed for it. We are planning to call
+  the feature `DynamicVolumeThreshold`.
+
+### API Changes
+
+We propose following changes to node's status field:
```

Comment:
Shouldn't this go under node.capacity with the other resource constraints?

Reply:
+1

Reply:
Drawback is that you have to request a resource type. But, according to @thockin, device plugin people are pushing a model where admission controller detects what type of volume you are requesting and it decorates the correct type of resource request. Agreed that we should try to repurpose this design.

Reply:
We had a discussion today in sig-node call and there were valid concerns raised against using `node.Capacity` for scheduling. The main problem being - there is no corresponding pod level field. memory, cpu and GPU devices are all specified at container level but volume is counted at pod level.

At this point we either heard some support for using `node.Status.AttachedVolumeLimits` or no strong opinion against. cc @derekwaynecarr 

If we choose to use node and pod capacity then we will have to create a new field in pod object spec and there was concern that storage might be only sub system using that field. We expect more comments as follow up.

Reply:
additional clarification:

"pods" never appears in a container resource requirement

if it did, it would be a problem.